tools/queries.cpp

```diff
@@ -326,32 +330,46 @@ int main(int argc, const char **argv)
-    }                                                                                    \
-    else if (app.index_encoding() == BOOST_PP_STRINGIZE(T))                              \
-    {                                                                                    \
-        if (app.is_wand_compressed()) {                                                  \
-            perftest<BOOST_PP_CAT(T, _index), wand_uniform_index>(app.index_basename(),  \
-                                                                  app.wand_data_path(),  \
-                                                                  app.queries(),         \
-                                                                  app.thresholds_file(), \
-                                                                  app.index_encoding(),  \
-                                                                  app.algorithm(),       \
-                                                                  app.k(),               \
-                                                                  app.scorer(),          \
-                                                                  extract,               \
-                                                                  safe);                 \
-        } else {                                                                         \
-            perftest<BOOST_PP_CAT(T, _index), wand_raw_index>(app.index_basename(),      \
-                                                              app.wand_data_path(),      \
-                                                              app.queries(),             \
-                                                              app.thresholds_file(),     \
-                                                              app.index_encoding(),      \
-                                                              app.algorithm(),           \
-                                                              app.k(),                   \
-                                                              app.scorer(),              \
-                                                              extract,                   \
-                                                              safe);                     \
+#define LOOP_BODY(R, DATA, T)                                                                \
+    }                                                                                        \
+    else if (app.index_encoding() == BOOST_PP_STRINGIZE(T))                                  \
+    {                                                                                        \
+        if (app.is_wand_compressed()) {                                                      \
+            if (quantized) {                                                                 \
+                perftest<BOOST_PP_CAT(T, _index), wand_uniform_index_quantized>(             \
```

Comment:
Actually, here's what you can do if you apply the same arguments to different functions depending on a condition:

```cpp
auto params = std::make_tuple(a, b, c, d, e);
...
if (a) { std::apply(f<A>, params); }
else if (b) { std::apply(f<B>, params); }
else if (c) { std::apply(f<C>, params); }
```